tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -2,42 +2,47 @@ import { useBackend } from '../backend';
 import { Button, DepartmentEntry, DepartmentPane, Icon, Stack } from '../components';
 import { Window } from '../layouts';
 import { Color } from 'common/color';
+import { merge } from 'common/merge';
 import { SFC } from 'inferno';
 import { JobToIcon } from './common/JobToIcon';
 import { BaseDepartmentInfo, BaseJobInfo } from '../components/DepartmentPane';
 
 class Job implements BaseJobInfo {
-  name: string;
   unavailable_reason: string | null;
   command: boolean;
   open_slots: number;
   used_slots: number;
   icon: string;
   prioritized: boolean;
-  job_description: string;
+  description: string;
 }
 
 class Department implements BaseDepartmentInfo {
-  name: string;
   color: string;
   open_slots: number;
-  jobs: Job[];
+  jobs: { [x: string]: Job };
 }
 
 type Data = {
-  departments: Department[];
+  static_data: { departments: { [x: string]: Department } };
+  departments: { [x: string]: Department };
   alert_state: string;
   shuttle_status: string;
   disable_jobs_for_non_observers: boolean;
   priority: boolean;
   round_duration: string;
 };
 
-// Specifically not typed for flexibility.
-export const JobEntry: SFC<{ job; department; act: Function }> = (data) => {
+export const JobEntry: SFC<{
+  jobName: string;
+  job: Job;
+  department: Department;
+  act: Function;
+}> = (data) => {
```

Comment:
Do not pass `act`, pass an `onClick` function instead.

```suggestion
type JobEntryProps = {
  jobName: string;
  job: Job;
  department: Department;
  onClick: () => void;
};

export const JobEntry = (props: JobEntryProps) => {
```